### ISSUE TITLE
Waterfox classic 2019.10 pops up window asking to stop unresponsive scripts instead of showing a message about website slowing it down

### ISSUE THREAD

**Issue by JulianRagan** [MrAlex94/Waterfox#1257]
Waterfox for some time has been showing an upper bar (similar to the one about blocking popup windows) informing about a webpage slowing it down and offering to stop it.

After updating to Waterfox 2019.10 classic it reverted to the old way of switching to the offending tab and showing a detailed dialog with option to stop an unresponsive script.

**Comment by grahamperrin** [comment]
Waterfox Classic 2019.10 (20191015002329) on Manjaro Linux at `https://github.com/zfsonlinux/zfs/pull/4329` (an extraordinarily long PR) with an extension that's quite hungry in a guest virtual machine throttled to 40% whilst the host machine was busy building software: 

![Warning: unresponsive script - modal](https://user-images.githubusercontent.com/192271/68085652-1af06b00-fe3b-11e9-8652-0d3f09c3a5cd.png)

Home-built Waterfox Classic 56.2.14 (20191007063316) on FreeBSD, the host: 

![image](https://user-images.githubusercontent.com/192271/68085814-d1088480-fe3c-11e9-896c-e16412c8a59d.png)

A modal dialogue with both versions of Waterfox Classic. 

I do vaguely recall occasionally seeing banners in lieu of modal dialogues, but I can't recall whether the banners were in Waterfox Classic.

**Comment by JulianRagan** [comment]
I have downgraded to 56.2.14 and then to 56.2.12 and I see that this behavior became new-old standard. I had to switch to a second user profile, which was not subject to Waterfox classic 2019.10 update. So I would assume this is due to user settings and not a bug in software. Below is what I meant by the upper bar
![SlowDown](https://user-images.githubusercontent.com/31183002/68088314-43d62780-fe5e-11e9-8c3f-37d087807a92.png)

Is there some easy way to compare user settings between profiles? I don't see a way to copy all about:config entries into say excel or export to csv.

**Comment by grahamperrin** [comment]
If it helps: my 56.2.14 screenshot above was of a profile that had never been touched by Waterfox Classic 2019.10. 

Where settings are in a `.json` file (in the _root_ directory of a profile): you can use Waterfox Classic to open and view the file. In the window: 

Raw data ▶ Pretty print

– maybe most useful if you need to seek and find a preference.

**Comment by grahamperrin** [comment]
Refreshed, safe mode: 

![image](https://user-images.githubusercontent.com/192271/68090178-94528280-fe68-11e9-8a4a-1ea6b7a31308.png)

**Comment by JulianRagan** [comment]
I have found an extension, config export, that lets me export about:config to csv file, but I can't find any difference that would point to this feature (I have been comparing 56.2.14, 56.2.12, 56.2.12 clean profile and 2019.10). So maybe it is not related to browser configuration.

**Comment by grahamperrin** [comment]
Create a new profile, start it in safe mode and run the test at https://browserbench.org/MotionMark1.1/ – expect slowness (from safety) …

**Comment by JulianRagan** [comment]
I did create a new profile on Waterfox Classic 2019.10, started it with -safe-mode command line parameter, selected safe mode and got the lovely modal window after a short while.
![Modal dialog](https://user-images.githubusercontent.com/31183002/68090958-5199a600-fe7a-11e9-908a-fb95499366bd.png)

So it seems to be random. I think (I am not sure), that at work I have observed both modal dialog and "nonmodal dialog upper bar" in a single session, but at work I am focused on work, so I am not certain of that.
